### ISSUE TITLE
Discussion: adding metrics to the register on the fly?

### ISSUE THREAD

**Issue by jeromedockes** [probabl-ai/skore#2903]
### Is your feature request related to a problem? Please describe.

for standard scikit-learn metric names, we could allow passing them to summarize even if they are not in the registry, and add them to the registry on the fly

### Describe the solution you'd like

even if `neg_mean_absolute_percentage_error` is not in the metrics registry, `report.summarize('neg_mean_absolute_percentage_error')` would work: it would add that metric to the registry then proceed as usual.

the signature of `.metrics.summarize` would not change.

**Comment by auguste-probabl** [comment]
If we do this we should also do it in `get`

**Comment by auguste-probabl** [comment]
And I think we should also do it for callables, not just sklearn strings, otherwise as a user I'd be surprised

**Comment by auguste-probabl** [comment]
In general I think this is a good proposal because the 2-step process is lengthy, but I'm also a bit reticent to make `summarize` and `get` stateful. There is a comfort in the guarantee that these methods will never mutate the report -- although now that I think about it, they do update the cache.
